rigour/ids/ogrn.py

```diff
@@ -25,3 +43,27 @@ def normalize(cls, text: str) -> Optional[str]:
         if match is None:
             return None
         return match.group(1)
+
+    @classmethod
+    def calculate_control_digit(cls, grn: str) -> Optional[int]:
+        if len(grn) == 13:
+            number = int(grn[:12])
+            mod_result = number % 11
+            calculated_digit = mod_result if mod_result != 10 else 0
+            return calculated_digit
+        elif len(grn) == 15:
+            number = int(grn[:14])
+            mod_result = number % 13
+            calculated_digit = mod_result if mod_result != 10 else 0
+            return calculated_digit
+        return None
+
+    @classmethod
+    def validate_control_digit(cls, grn: str) -> bool:
+        if len(grn) == 13:
+            control_digit = int(grn[12])
+            return control_digit == cls.calculate_control_digit(grn)
+        elif len(grn) == 15:
+            control_digit = int(grn[14])
+            return control_digit == cls.calculate_control_digit(grn)
```

Comment:
would control_digit = grn[-1] give the same result for both legnths?

No need to change, this is fine, but just an alternative